### ISSUE TITLE
How to generate the npy file and hdf file of training_set

### ISSUE THREAD

**Issue by xuelianl** [ICOA-SBC/MD_DL_BA#5]
Hi,

Thank you for sharing this interesting work!
I have two questions regarding the data preprocessing:
- Would it be possible to share the scripts on how to generate the numpy file(.npy) or hdf file of training/validation/test set from MD data(trajactory)?
- What is the purpose of charge normalization in training densenucy?

Thank you for your time and consideration.

**Comment by xuelianl** [comment]
@cparisel @shalkya @MaximeSongIdris @SBC-ICOA 
If you have a moment to look at this, I'd greatly appreciate!

**Comment by shalkya** [comment]
Hello @xuelianl, thank you for your interest in this work.
I don't currently have access to these scripts, and I have already reached out to the current owner so that I can add them in the github repo.

In the meantime, you can checkout the initial code from Pafnucy for creating HDF5: https://gitlab.com/cheminfIBB/pafnucy/-/blob/master/pdbbind_data.ipynb?ref_type=heads

The code, in the previous link, had to be adapted to create the datasets from the frames of MD simulations.


As for the charge normalization, I refer you to the publication of Pafnucy:
"The partial charges were scaled by the training set’s standard deviation in order to get a distribution with a unit standard deviation, which improves learning."
https://academic.oup.com/bioinformatics/article/34/21/3666/4994792

**Comment by xuelianl** [comment]
@shalkya Thank you for the quick and valuable response! I will go to check the paper and code of Pafnucy first.

**Comment by shalkya** [comment]
@xuelianl I have added the 2 scripts used to create HDF5 and .npy datasets.
Please bear in mind that, initially, I didn't plan to provide these scripts on the github repository.
I had to update them so that other users could run them. Unfortunately, I don't have access to the data (frames etc) to test these updated versions, so you will have to debug the code to some extend.
Lastly, there are several steps between performing the MD simulations and creating these datasets, such as extracting the frames and calculating the pockets, as well as converting the pocket files into .mol2 files (these steps are not covered in the provided scripts, I advise you to use tools such as pytraj and MDAnalysis for these tasks).

Nonetheless, I hope these scripts will be useful to you.

**Comment by xuelianl** [comment]
Thank you so much for providing these scripts, which have already provided more than enough information, and they are very important for me to reproduce your work!
